libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -362,6 +362,8 @@ bool ASTJsonConverter::visit(FunctionDefinition const& _node)
 		make_pair("implemented", _node.isImplemented()),
 		make_pair("scope", idOrNull(_node.scope()))
 	};
+	if (_node.isPartOfExternalInterface())
+		attributes.emplace_back("functionSelector", toCompactHexWithPrefix(_node.externalIdentifier()));
```

Comment:
compact does also not help, it might leave out the first byte if it is zero.

Reply:
Maybe it would be best to add another function similar to `externalIdentifier` to `FunctionType` that returns the hex value:
```
string FunctionType::externalIdentifierHex() const
{
	return FixedHash<4>(dev::keccak256(externalSignature())).hex();
}
```

Reply:
This function doesn't add "0x" prefix, should I add it, or keep the value without the prefix?

Also should I keep the functions `FunctionDeclaration::externalIdentifier()` and `VariableDeclaration::externalIdentifier()`? They are not used anywhere, so I though that I'd replace them with `FunctionDeclaration::externalIdentifierHex()` and `VariableDeclaration::externalIdentifierHex()`.

Reply:
If they are not used anywhere, then please replace them.

About the `0x` prefix: I do not think it is needed.

Reply:
Done here 62cb90c3a2c7f16478713f5068ebbfcf8bc91e8c